clients/python-wrapper/lakefs/object.py

```diff
@@ -676,13 +676,15 @@ def reader(self, mode: ReadModes = 'rb', pre_sign: Optional[bool] = None) -> Obj
         """
         return ObjectReader(self, mode=mode, pre_sign=pre_sign, client=self._client)
 
-    def stat(self) -> ObjectInfo:
+    def stat(self, pre_sign: Optional[bool] = None) -> ObjectInfo:
```

Comment:
1. Because we use named parameters, consider using the name param name 'presign'.
2. Why not include user metadata and close the issue?

Reply:
convolutes the caching behavior. If I cached an object with no user_metadata and now I request it with user_metadata, I need to make sure not to use the cached version (and vice versa).

As for `pre_sign` vs `presign` - I decided  to adopt the style of the surrounding code (see for example the `reader()` method above)